src/utils/service/service.js

```diff
@@ -160,35 +161,85 @@ class Service {
+   * @return {Object}
+   */
+  axiosParams(options) {
+    delete options.onSuccess;
+    delete options.onError;
+    return options;
+  }
+
+  /**
    * Performs a GET request to the server.
    *
    * @method get
    * @param {Number} id - the ID of the resource.
-   * @param {Function} onSuccess - a callback to trigger on success
-   * @param {Function} onError - a callback to trigger on error
+   * @param {Object} options - an object with options for query params and success/error callbacks
+   * @param {Function} _onError - [Deprecated] a callback to trigger on error
    * @return {Void}
    */
-  get(id, onSuccess, onError) {
-    this.client.get(String(id)).then(
-      this.handleResponse.bind(this, onSuccess),
-      this.handleResponse.bind(this, onError)
-    );
+  get(id, options, _onError) {
+    let request;
+
+    if (typeof (options) === 'function') {
+      Logger.deprecate('Passing onSuccess and onError as separate arguments is deprecated. Please use `get(1, { onSuccess: function, onError: function, params: { k: "v" } })`'); // eslint-disable-line max-len
+
+      request = this.client.get(String(id));
+
+      request.then(
+        this.handleResponse.bind(this, options),
+        this.handleResponse.bind(this, _onError)
+      );
+    } else {
+      const onSuccess = options.onSuccess;
+      const onError = options.onError;
+      const params = this.axiosParams(options);
+
+      request = this.client.get(String(id), params);
+      request.then(
+        this.handleResponse.bind(this, onSuccess),
+        this.handleResponse.bind(this, onError)
+      );
+    }
   }
 
   /**
    * Performs a POST request to the server.
    *
    * @method post
    * @param {Object} data - the data to post to the server.
-   * @param {Function} onSuccess - a callback to trigger on success
-   * @param {Function} onError - a callback to trigger on error
+   * @param {Object} options - an object with options for query params and success/error callbacks
+   * @param {Function} _onError - [Deprecated] a callback to trigger on error
    * @return {Void}
    */
-  post(data, onSuccess, onError) {
-    this.client.post('', data).then(
-      this.handleResponse.bind(this, onSuccess),
-      this.handleResponse.bind(this, onError)
-    );
+  post(data, options, _onError) {
+    let request;
+
+    if (typeof (options) === 'function') {
+      Logger.deprecate("Passing onSuccess and onError as separate arguments is deprecated. Please use `post('', { data_key: 'data_val' }, { onSuccess: function, onError: function, params: { k: 'v' } })`"); // eslint-disable-line max-len
+      request = this.client.post('', data);
+
+      request.then(
+        this.handleResponse.bind(this, options),
+        this.handleResponse.bind(this, _onError)
+      );
+    } else {
+      const onSuccess = options.onSuccess;
+      const onError = options.onError;
+      const params = this.axiosParams(options);
```

Comment:
sorry - just remembered about destructuring assignments which would be really useful here

```js
const { onSuccess, onError, ...params } = options;
```

it would clean the code up a little bit